src/main/scala/is/hail/keytable/KeyTable.scala

```diff
@@ -62,44 +32,40 @@ object KeyTable {
   }
 }
 
-case class KeyTable(@transient hc: HailContext, @transient rdd: RDD[(Annotation, Annotation)],
-  keySignature: TStruct, valueSignature: TStruct) extends Serializable {
-  require(fieldNames.areDistinct())
-
-  private val localValueSignature = valueSignature
-  val (signature, mergeKeyAndValue) = keySignature.merge(localValueSignature)
+case class KeyTable(@transient hc: HailContext, @transient rdd: RDD[Annotation],
+  signature: TStruct, keyNames: Array[String]) {
+  require(fieldNames.areDistinct() && keyNames.areDistinct())
+  require(keyNames.forall(fieldNames.contains(_)))
 
   def fields = signature.fields
 
-  def keySchema = keySignature.schema
-
-  def valueSchema = valueSignature.schema
+  def keyFields = keyNames.flatMap { n => signature.fieldOption(n) }
```

Comment:
I wanted the fields of the keys in the order of keyNames. I didn't see a method on signature to get a field based on name without it being Option[Field]. I rewrote it to use the index to get the fields directly.